tools/codegen/core/gen_upb_api.sh

```diff
@@ -124,3 +124,18 @@ do
     --upb_out=$UPB_OUTPUT_DIR \
     --plugin=protoc-gen-upb=$UPB_PLUGIN
 done
+
+# In PHP build Makefile, the files with .upb.c suffix collide .upbdefs.c suffix due to a PHP buildsystem bug.
+# Work around this by placing the generated files with ".upbdefs.c" suffix under a different directory.
+# See https://github.com/grpc/grpc/issues/23307
+
+# move all .upbdefs.c files from under src/core/ext/upb-generated to src/core/ext/upbdefs-generated
+UPBDEFS_OUTPUT_DIR=$PWD/src/core/ext/upbdefs-generated
```

Comment:
If we wind up going with this approach, you will also need to change tools/distrib/check_upb_output.sh to check the upbdefs directory as well.

Reply:
I will check the situation with #23122 and come back with an opinion.